tests/cloud_functions/testing_utils.py

```
@@ -0,0 +1,224 @@
+
+class InMemoryFirestoreDoc:
+    """A class that simulates a Firestore document in memory.
+
+    Attributes:
+        _data (dict): The data stored in the document.
+        exists (bool): Indicates whether the document exists.
+        _subcollections (defaultdict): A dictionary of subcollections,
+            where each subcollection is an instance of InMemoryFirestoreCollection.
+    """
+
+    def __init__(self, data=None):
```

Ruff (0.8.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)